services/slam/fake/slam.go

```diff
@@ -0,0 +1,160 @@
+// Package fake implements a fake base.
+package fake
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"image"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/edaniels/golog"
+	"github.com/golang/geo/r3"
+	"go.viam.com/utils"
+	"go.viam.com/utils/artifact"
+
+	"go.viam.com/rdk/components/generic"
+	"go.viam.com/rdk/config"
+	"go.viam.com/rdk/pointcloud"
+	"go.viam.com/rdk/referenceframe"
+	"go.viam.com/rdk/registry"
+	"go.viam.com/rdk/resource"
+	"go.viam.com/rdk/rimage"
+	"go.viam.com/rdk/services/slam"
+	"go.viam.com/rdk/spatialmath"
+	rdkutils "go.viam.com/rdk/utils"
+	"go.viam.com/rdk/vision"
+)
+
+var model = resource.NewDefaultModel("fake")
+
+const (
+	internalStateTemplate = "slam/example_cartographer_outputs/internal_state/internal_state_%d.pbstream"
+	maxDataCount          = 16
+	pcdTemplate           = "slam/example_cartographer_outputs/pointcloud/pointcloud_%d.pcd"
+	pngTemplate           = "slam/example_cartographer_outputs/image_map/image_map_%d.png"
+	positionTemplate      = "slam/example_cartographer_outputs/position/position_%d.txt"
+)
+
+func init() {
+	registry.RegisterService(
+		slam.Subtype,
+		model,
+		registry.Service{
+			Constructor: func(
+				ctx context.Context,
+				_ registry.Dependencies,
+				config config.Service,
+				logger golog.Logger,
+			) (interface{}, error) {
+				return &SLAM{Name: config.Name, dataCount: 1}, nil
+			},
+		},
+	)
+}
+
+var _ = slam.Service(&SLAM{})
+
+// SLAM is a fake slam that returns generic data.
+type SLAM struct {
+	generic.Echo
+	Name      string
+	dataCount int
+}
+
+func (slamSvc *SLAM) GetMap(ctx context.Context, name, mimeType string, cp *referenceframe.PoseInFrame,
+	include bool, extra map[string]interface{},
+) (string, image.Image, *vision.Object, error) {
+	var err error
+	var img image.Image
+	var vObj *vision.Object
+
+	// Increment data after getMap call
+	slamSvc.incrementDataCount()
+
+	switch mimeType {
+	case rdkutils.MimeTypePCD:
+		f, err := os.Open(artifact.MustPath(fmt.Sprintf(pcdTemplate, slamSvc.dataCount)))
+		if err != nil {
+			return "", nil, nil, err
+		}
+		defer utils.UncheckedErrorFunc(f.Close)
+		pc, err := pointcloud.ReadPCDToBasicOctree(f)
```

Comment:
The old GetMap isn't expected to return an octree.

Reply:
I don't think it is. GetMap is returning a `*vision.Object` if the mime-type is `pointcloud/pcd`.  

vision.NewObject only requires its input to fulfill the pointecloud interface (which both KDTrees & Octrees do). 

I was under the impression that the acceptance criteria for the Octree work is that Octree & KDTree values created from the same PCD file are indistinguishable (other than space & runtime performance) when interacted with via the PointCloud interface methods alone (which I think is what vision.NewObject does. 

My impression is that you would get functionally the same vObj value if you did 
```
pc, _ := pointcloud.ReadPCDToKDTree(f)
vObj, _ = vision.NewObject(pc)
```

or 
```
pc, _ := pointcloud.ReadPCDToBasicOctree(f)
vObj, _ = vision.NewObject(pc)
```

Is that not true?

Reply:
Yes, it's fine to return an Octree here, but it's not what the current GetMap does, so if we're trying to mimic the current GetMap, then I would not create an octree here. But since it does fulfill the PointCloud interface and GetMap is also going away, this isn't that important, so feel free to resolve.

Reply:
@tessavitabile On further investigation... I think you might have caught a bug. 

I changed this function to use `ReadPCD` rather than `ReadPCDToBasicOctree` which fixes the issue with the frontend PCD viewer not being able to render the PCD.

Does that mean that our octree implementation has a bug & that I should create a ticket to investigate?

Reply:
Yes, that's a good idea. Good find!